locust/test/test_dispatch.py

```diff
@@ -3291,6 +3291,241 @@ class User3(User):
+
+    def case_handler(self, cases: List[Case], expected: Dict[str, int], user_classes: List[User]):
+        self.assertEqual(len(cases), len(expected))
+
+        for case_num in range(len(cases)):
+            # Reset to defaul values
+            for user_class in user_classes:
+                user_class.weight, user_class.fixed_count = 1, 0
+
+            case = cases[case_num]
+            self.assertEqual(
+                len(case.fixed_counts) + len(case.weights),
+                len(user_classes),
+                msg="Invalid test case or user list.",
+            )
+
+            fixed_users = user_classes[: len(case.fixed_counts)]
+            weighted_users_list = user_classes[len(case.fixed_counts) :]
+
+            for user, fixed_count in zip(fixed_users, case.fixed_counts):
+                user.fixed_count = fixed_count
+
+            for user, weight in zip(weighted_users_list, case.weights):
+                user.weight = weight
+
+            worker_node1 = WorkerNode("1")
+
+            users_dispatcher = UsersDispatcher(worker_nodes=[worker_node1], user_classes=user_classes)
+            users_dispatcher.new_dispatch(target_user_count=case.target_user_count, spawn_rate=0.5)
+            users_dispatcher._wait_between_dispatch = 0
+
+            iterations = list(users_dispatcher)
+            self.assertDictEqual(iterations[-1]["1"], expected[case_num])
+
+    def test_ramp_up_2_weigted_user_with_1_fixed_user(self):
+        class User1(User):
+            ...
+
+        class User2(User):
+            ...
+
+        class User3(User):
+            ...
+
+        self.case_handler(
+            cases=[
+                self.Case(fixed_counts=(1,), weights=(1, 1), target_user_count=3),
+                self.Case(fixed_counts=(1,), weights=(1, 1), target_user_count=9),
+                self.Case(fixed_counts=(8,), weights=(1, 1), target_user_count=10),
+                self.Case(fixed_counts=(2,), weights=(1, 1), target_user_count=1000),
+                self.Case(fixed_counts=(100,), weights=(1, 1), target_user_count=1000),
+                self.Case(fixed_counts=(960,), weights=(1, 1), target_user_count=1000),
+                self.Case(fixed_counts=(9990,), weights=(1, 1), target_user_count=10000),
+                self.Case(fixed_counts=(100,), weights=(1, 1), target_user_count=100),
+            ],
+            expected=[
+                {"User1": 1, "User2": 1, "User3": 1},
+                {"User1": 1, "User2": 4, "User3": 4},
+                {"User1": 8, "User2": 1, "User3": 1},
+                {"User1": 2, "User2": 499, "User3": 499},
+                {"User1": 100, "User2": 450, "User3": 450},
+                {"User1": 960, "User2": 20, "User3": 20},
+                {"User1": 9990, "User2": 5, "User3": 5},
+                {"User1": 100, "User2": 0, "User3": 0},
+            ],
+            user_classes=[User1, User2, User3],
+        )
+
+    def test_ramp_up_various_count_weigted_and_fixed_users(self):
```

Comment:
It could also be a good to add a test for ramping-down a mix of weighted and fixed users (i.e. `test_ramp_down_various_count_weigted_and_fixed_users`). If you implement this ramp-down test in the current test class, make sure to also rename the class to something like `TestRampUpAndDownUsersWithFixedCound`

Reply:
Yeah, you're right.
The name changed.